### PR TITLE
Feature/zipkin2 support

### PR DESCRIPTION
**Pull request by nicmunroe**

### CODE REVIEW

**Comment by nicmunroe** [comment]
@adriancole This builds on your commits in #69 so that Wingtips uses the native Zipkin `Reporter`s and `Sender`s rather than the custom Wingtips `ZipkinSpanSender` adapter. I'd love to get your thoughts before moving forward with this if you have time to give it a quick look.

File: wingtips-zipkin2-spring-boot/src/main/java/com/nike/wingtips/springboot/zipkin2/WingtipsZipkinProperties.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * A {@link ConfigurationProperties} companion for {@link WingtipsWithZipkinSpringBootConfiguration} that allows you to
+ * specify the configuration of {@link WingtipsToZipkinLifecycleListener} via your Spring Boot application's properties
+ * files. The following properties are supported (NOTE: {@code wingtips.zipkin.base-url} is required - all others are
+ * optional and can be left out, however it's highly recommended that you set {@code wingtips.zipkin.service-name} as
+ * well):
+ * <ul>
+ *     <li>
+ *         wingtips.zipkin.zipkin-disabled - Disables registering {@link WingtipsToZipkinLifecycleListener} with
```
Comment (adriancole) [inline wingtips-zipkin2-spring-boot/src/main/java/com/nike/wingtips/springboot/zipkin2/WingtipsZipkinProperties.java]: probably not great idea to change, but I think in boot it is conventional to just end in "enabled"

**Comment by adriancole** [comment]
thanks, Nic! Appreciate your taking the time here

File: wingtips-zipkin2-spring-boot/src/main/java/com/nike/wingtips/springboot/zipkin2/WingtipsZipkinProperties.java (hunk 1 above)
Comment (nicmunroe) [reply wingtips-zipkin2-spring-boot/src/main/java/com/nike/wingtips/springboot/zipkin2/WingtipsZipkinProperties.java]: If I had a time machine I'd go back and do it right the first time, but yeah, probably not a good idea to change now. :)

File: wingtips-zipkin2-spring-boot/src/main/java/com/nike/wingtips/springboot/zipkin2/WingtipsZipkinProperties.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ *     wingtips.zipkin.service-name=some-service-name
+ * </pre>
+ *
+ * @author <a href="mailto:ales.justin@jboss.org">Ales Justin</a>
+ */
+@ConfigurationProperties("wingtips.zipkin")
+@SuppressWarnings("WeakerAccess")
+public class WingtipsZipkinProperties {
+    private boolean zipkinDisabled = false;
+    private String serviceName = "unknown";
```
[the remark this answers is not in the document]
Comment (nicmunroe) [reply wingtips-zipkin2-spring-boot/src/main/java/com/nike/wingtips/springboot/zipkin2/WingtipsZipkinProperties.java]: I added a todo in the code to investigate this later. I don't want to dive into the spring weeds to figure out how to safely pull in the `spring.application.name` property in a clean way right now.

File: wingtips-zipkin2-spring-boot/src/main/java/com/nike/wingtips/springboot/zipkin2/WingtipsWithZipkinSpringBootConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    public WingtipsWithZipkinSpringBootConfiguration(WingtipsZipkinProperties wingtipsZipkinProperties) {
+        this.wingtipsZipkinProperties = wingtipsZipkinProperties;
+        init();
+    }
+
+    /**
+     * Initialize configuration.
+     * Add Zipkin listener if our {@link WingtipsZipkinProperties} indicates it has the necessary properties specified.
+     */
+    private void init() {
```
[the remark this answers is not in the document]
Comment (nicmunroe) [reply wingtips-zipkin2-spring-boot/src/main/java/com/nike/wingtips/springboot/zipkin2/WingtipsWithZipkinSpringBootConfiguration.java]: Great idea. Done. Users can now define a `Reporter<zipkin2.Span>` bean in their app's Spring config and we'll use that instead. If no such `Reporter` bean is defined then we'll use the default `AsyncReporter` with `URLConnectionSender` behavior.

File: wingtips-zipkin2/src/main/java/com/nike/wingtips/zipkin2/WingtipsToZipkinLifecycleListener.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+     * @param serviceName The name of this service. This is used to build the Zipkin {@link Endpoint}, which tells
+     * Zipkin which service generated the spans we're going to send it.
+     * @param zipkinSpanConverter The {@link WingtipsToZipkinSpanConverter} that should be used to convert Wingtips
+     * spans to Zipkin spans.
+     * @param zipkinSpanReporter The Zipkin {@link Reporter} for collecting and sending Zipkin spans to the Zipkin
+     * server.
+     */
+    public WingtipsToZipkinLifecycleListener(String serviceName, WingtipsToZipkinSpanConverter zipkinSpanConverter, Reporter<zipkin2.Span> zipkinSpanReporter) {
+        this.serviceName = serviceName;
+        this.zipkinEndpoint = Endpoint.newBuilder().serviceName(serviceName).build();
```
[the remark this answers is not in the document]
Comment (nicmunroe) [reply wingtips-zipkin2/src/main/java/com/nike/wingtips/zipkin2/WingtipsToZipkinLifecycleListener.java]: Added a todo for this. Thanks for the idea, and the reference to the `Platform` code I can pull from. I'm assuming you're suggesting adding `builder.ip(...)` (not using the IP as the service name), and I'm assuming you're referring to this code: https://github.com/openzipkin/brave/blob/af055a61330a10afa9b6fa4f05f7d33a3b3a7296/brave/src/main/java/brave/internal/Platform.java#L60-L83. Yes/no?

**Comment by nicmunroe** [comment]
Thanks for the review Adrian!

**Pull request merged**